c7n/query.py

```diff
@@ -551,14 +548,13 @@ def check_resource_limit(self, selection_count, population_count):
 
     def _get_cached_resources(self, ids):
         key = self.get_cache_key(None)
-        if self._cache.load():
+        with self._cache:
```

Comment:
Nice, didn't notice that you already had the logic in place to load on enter and close on exit. This is clean :+1: